cicecore/shared/ice_fileunits.F90

```diff
@@ -116,7 +116,11 @@ subroutine init_fileunits
          ice_IOUnitsInUse(ice_stdout) = .true. ! reserve unit 6
          ice_IOUnitsInUse(ice_stderr) = .true.
          if (nu_diag >= 1 .and. nu_diag <= ice_IOUnitsMaxUnit) &
-            ice_IOUnitsInUse(nu_diag) = .true. ! reserve unit nu_diag
+              ice_IOUnitsInUse(nu_diag) = .true. ! reserve unit nu_diag
+#ifdef CESMCOUPLED
+         ! CESM can have negative unit numbers.
+         if (nu_diag < 0) nu_diag_set = .true.
```

Comment:
Had a quick look at "nu_diag_set" usage and don't love it overall.  Can we put #ifdef CESMCOUPLED" around it everywhere, so it's clear it has nothing to do with CICE.

Reply:
including in the declaration in ice_fileunits.F90.

Reply:
or could we get rid of nu_diag_set and use CESMCOUPLED cpp to "turn off" whatever CICE is doing to set/release nu_diag?  With CESMCOUPLED, if we can assume nu_diag is set by the coupler layer, we shouldn't need "nu_diag_set".

Reply:
This is fine.